lib/src/rules/newline_to_start_multiline_strings.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Start multiline strings with a newline.';
+
+const _details = r'''
+
+Multiline strings are easier to read when they start with a newline (a newline
+starting a multiline string is ignored).
+
+**BAD:**
+```
+var s1 = """{
+  "a": 1,
+  "b": 2
+}""";
+```
+
+**GOOD:**
+```
+var s1 = """
+{
+  "a": 1,
+  "b": 2
+}""";
+
+var s2 = """This onliner multiline string is ok. It usually allows to escape both ' and " in the string.""";
+```
+
+''';
+
+class NewlineToStartMultilineStrings extends LintRule implements NodeLintRule {
+  NewlineToStartMultilineStrings()
+      : super(
+            name: 'newline_to_start_multiline_strings',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(NodeLintRegistry registry,
+      [LinterContext context]) {
+    final visitor = _Visitor(this);
+    registry.addCompilationUnit(this, visitor);
+  }
+}
+
+class _Visitor extends RecursiveAstVisitor<void> {
```

Comment:
This should extend `SimpleAstVisitor` and register for the other visit methods that need to be called. The way it's currently written we'll end up visiting the AST structure twice, which is not very efficient.